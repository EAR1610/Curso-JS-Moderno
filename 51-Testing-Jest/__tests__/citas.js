import Citas from '../js/classes/Citas';

describe('Probar la clase de Citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Alexander',
            telefono: '566655',
            fecha: '2/8/2001',
            hora:'12:50',
            sintomas: 'Solo duerme'
        };

        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Alexander',
            telefono: '566655',
            fecha: '2/8/2001',
            hora:'12:50',
            sintomas: 'Solo duerme'
        };     
        
        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
      citas.eliminarCita(id);

      expect(citas).toMatchSnapshot();
    });
})