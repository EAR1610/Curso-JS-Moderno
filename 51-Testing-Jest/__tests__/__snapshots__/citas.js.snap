// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "2/8/2001",
      "hora": "12:50",
      "id": 1619971643404,
      "mascota": "Nuevo nombre",
      "propietario": "Alexander",
      "sintomas": "Solo duerme",
      "telefono": "566655",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "2/8/2001",
      "hora": "12:50",
      "id": 1619971643404,
      "mascota": "Hook",
      "propietario": "Alexander",
      "sintomas": "Solo duerme",
      "telefono": "566655",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
